### PR TITLE
Update Dockerfile

### PR DESCRIPTION
**Pull request by timja**
Signed-off-by: Tim Jacomb <21194782+timja@users.noreply.github.com>

## Hey, I just made a Pull Request!

- Give node user ownership of yarn files recently added to backend Docker image so that modifying cache works
- Give node user ownership of WORKDIR so that untarring files works

#### :heavy_check_mark: Checklist

- ~[ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))~
- ~[ ] Added or updated documentation~
- ~[ ] Tests for new functionality and regression tests for bug fixes~
- ~[ ] Screenshots attached (for UI changes)~
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/backend/Dockerfile [hunk 1]
```diff
@@ -19,15 +19,17 @@ RUN apt-get update && \
     yarn config set python /usr/bin/python3
 
 # From here on we use the least-privileged `node` user to run the backend.
-USER node
 WORKDIR /app
+RUN chown node:node /app
```
Comment (timja) [inline packages/backend/Dockerfile:23]: without this I got:

```
 => ERROR [ 7/10] RUN tar xzf skeleton.tar.gz && rm skeleton.tar.gz                                                                                                                                                                                                      0.2s
------
 > [ 7/10] RUN tar xzf skeleton.tar.gz && rm skeleton.tar.gz:
#11 0.155 tar: packages: Cannot mkdir: Permission denied
#11 0.155 tar: packages/backend/package.json: Cannot open: No such file or directory
#11 0.155 tar: packages: Cannot mkdir: Permission denied
#11 0.155 tar: packages/app/package.json: Cannot open: No such file or directory
#11 0.155 tar: Exiting with failure status due to previous errors
```

File: packages/backend/Dockerfile [hunk 2]
```diff
@@ -19,15 +19,17 @@ RUN apt-get update && \
...
+USER node
+
 
 # This switches many Node.js dependencies to production mode.
 ENV NODE_ENV production
 
 # Copy over Yarn 3 configuration, release, and plugins
-COPY .yarn ./.yarn
-COPY .yarnrc.yml ./
```
Comment (timja) [inline packages/backend/Dockerfile:30]: Without this I got:
```
 => ERROR [ 9/11] RUN yarn workspaces focus --all --production && rm -rf "$(yarn cache clean)"                                                                                                                                                                           0.7s
------
 > [ 9/11] RUN yarn workspaces focus --all --production && rm -rf "$(yarn cache clean)":
#13 0.674 Internal Error: EACCES: permission denied, mkdir '/app/.yarn/cache'
#13 0.674 Error: EACCES: permission denied, mkdir '/app/.yarn/cache'
```

**Review by timja — COMMENTED** [review]
based on my trying to upgrade to yarn v3 in my own app

File: packages/backend/Dockerfile [hunk 3]
```diff
@@ -19,15 +19,17 @@ RUN apt-get update && \
     yarn config set python /usr/bin/python3
 
 # From here on we use the least-privileged `node` user to run the backend.
-USER node
 WORKDIR /app
+RUN chown node:node /app
+USER node
```
Comment (planeiii-te) [inline packages/backend/Dockerfile:24]: What's the benefit/goal of moving this? What are you trying to accomplish? 

FYI usually the mission statement is summarized in the PR title and greater detail is providing in the opening comment.

Comment (timja) [reply packages/backend/Dockerfile:24]: It’s in the self-review if you see my comments

Comment (planeiii-te) [reply packages/backend/Dockerfile:24]: This isn't an either-or situation though. `Update Dockerfile` is a pretty meaningless PR title. Anyone can see from you editing the file that it has been updated. What is your end goal though? What does updating it fix/improve/accomplish? You can edit the title and opening comment to make them both more meaningful.

**Review by freben — APPROVED** [review]
Alright!

File: packages/backend/Dockerfile (hunk 1 above)
Comment (Rugvip) [reply packages/backend/Dockerfile:23]: Is that with the reordering of the user command? I think we should be fine as long as you declare the user before the workdir?

Comment (timja) [reply packages/backend/Dockerfile:23]: Yes, it is with the reordering, the untar wasn’t working for me without this

Comment (timja) [reply packages/backend/Dockerfile:23]: seems to, see https://github.com/backstage/backstage/pull/13747

**Pull request merged**